File: 01768-merge-strings-alternately/Solution.php
<?php

declare(strict_types=1);

class Solution {

    /**
     * @param string $word1
     * @param string $word2
     * @return string
     */
    function mergeAlternately(string $word1, string $word2): string {

        $len = max(strlen($word1), strlen($word2));
        $res = '';
        
        for($i = 0; $i < $len; $i++) {
            
            $res .= $word1[$i] . $word2[$i];
        }
        
        return $res; 
    }
}

File: 01768-merge-strings-alternately/solution.go
func mergeAlternately(word1 string, word2 string) string {

	n := len(word1)
	m := len(word2)
	var res string

	for i := 0; i < n || i < m; i++ {

		if i < n {
			res += string(word1[i])
		}

		if i < m {
			res += string(word2[i])
		}
	}

	return res
}

// 0ms Beats 100.00% of users with Go, 2.82 MB Beats 12.72% of users with Go
